check.py

```diff
@@ -93,43 +121,44 @@ def setup_waterfall():
   except Exception, e:
     warn('could not run vanilla LLVM from waterfall: ' + str(e) + ', looked for clang at ' + CLANG)
 
-fetch_waterfall()
-setup_waterfall()
+if options.test_waterfall:
+  fetch_waterfall()
+  setup_waterfall()
 
-if only_prepare:
+if options.only_prepare:
   print 'waterfall is fetched and setup, exiting since --only-prepare'
   sys.exit(0)
 
 # external tools
 
 has_node = False
 try:
-  subprocess.check_call(['nodejs', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  subprocess.check_call(['nodejs', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
```

Comment:
We are doing similar Windows work on the waterfall script currently as well, as we're bringing up mac and Windows builds, and I was just looking at a similar change there.  Our script infrastructure has a few batch files which we found we needed `shell=True` for that; but I don't think any of these are batch files, and I'd rather not have `shell=True` if we don't have to. I assume it's in this PR for a reason though, what makes it necessary here?